File: modules/core/mgmt/rwrestconf/test/samples/ws_stream_client.py
#!/usr/bin/python3
import tornado.websocket as ws
import tornado.gen
import tornado.ioloop
import tornado.httpclient

# For XML content
#WS_URL = "ws://127.0.0.1:8888/ws_streams/NETCONF"

# For Json content
WS_URL = "wss://127.0.0.1:8888/ws_streams/NETCONF-JSON"
HTTP_REQ = tornado.httpclient.HTTPRequest(
              url = WS_URL,
              auth_username = "admin",
              auth_password = "admin",
              validate_cert = False)


@tornado.gen.coroutine
def websocket_client():
  websock = yield ws.websocket_connect(HTTP_REQ)
  print('=== Websocket connected')
  while True:
    msg = yield websock.read_message()
    if msg is None: break
    print("Received Message: {}".format(msg))


tornado.ioloop.IOLoop.current().run_sync(websocket_client)

File: modules/core/mgmt/rwrestconf/test/samples/ws_stream_client.js
#!/usr/bin/node

// Requires the websocket node module , node version >=0.10.32
var WebSocket = require('websocket').w3cwebsocket;
 
//var client = new WebSocket('ws://localhost:8888/ws_streams/NETCONF');
var client = new WebSocket('ws://localhost:8888/ws_streams/NETCONF-JSON');
 
client.onerror = function() {
    console.log('Connection Error');
};
 
client.onopen = function() {
    console.log('WebSocket Client Connected');
};
 
client.onclose = function() {
    console.log('Websocket Client Closed');
};
 
client.onmessage = function(e) {
    console.log("onmessage==");
    if (typeof e.data === 'string') {
        console.log("Received: '" + e.data + "'");
    }
};
